### PR TITLE
Added a simple HTML report and a simplified visual representation

### PR DESCRIPTION
**Pull request by DaJansenGit**
This creates a simple model report to solve #683 

To test it, just execute `e2_export_aixlib_models.py`. The report is deactivated by default and can be activated by setting `report=True` in `prj.export_aixlib()` function.

If activated, the report will be exported into a new subfolder `ModelReport` inside the `Resources` folder of the modelica exported mode.

There are two types of report:
* CSV export
* HTML export which also includes a heavy abstracted 3D visualization of the created buildings.

Here you can find a Screenshot of a report of an archetype building:
image

### CODE REVIEW

**Comment by MartinRaetz** [comment]
Kompatibilität:

Chrome works.

Firefox:
grafik

File: teaser/data/output/reports/model_report.py [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+            area_east=prj_data_flat["Outerwall Area_90.0"],
+            area_south=prj_data_flat["Outerwall Area_180.0"],
+            area_west=prj_data_flat["Outerwall Area_270.0"],
+            height=prj_data_flat["Floor Height"],
+            window_area_north=prj_data_flat["Window Area_0.0"],
+            window_area_east=prj_data_flat["Window Area_90.0"],
+            window_area_south=prj_data_flat["Window Area_180.0"],
+            window_area_west=prj_data_flat["Window Area_270.0"],
+            num_floors=prj_data_flat["Number of Floors"],
+            roof_angle=30)
```
Comment (MartinRaetz) [inline teaser/data/output/reports/model_report.py:225]: hard coded roof_angle. Will this lead to problems in future?

**Review by MartinRaetz — APPROVED** [review]
See the comments also.

I do think the proposed solution is suitable for merging. However, feel free to address the proposed improvements or directly merge.

**Comment by MartinRaetz** [comment]
**Feedback**
Extremely helpful feature! And a very nice implemenation!

**Suggestions:**
- legend: grey = walls, blue = windows; 0° = south; 90°= west



**Questions:**

- U values are mean U-values?
- The values shown upon hovering over plot like Trace45 have meaning?
- why only for AixLib Export? Should other exports have the same feature?

**Comment by MartinRaetz** [comment]
Upon using other building types the following error occurs and suggests that non existent walls/windows need to be handled:
This test used example 6 to be exported.

`Traceback (most recent call last):
  File "D:\04_GitRepos\TEASER\teaser\examples\e2_export_aixlib_models.py", line 81, in <module>
    example_export_aixlib()
  File "D:\04_GitRepos\TEASER\teaser\examples\e2_export_aixlib_models.py", line 71, in example_export_aixlib
    path = prj.export_aixlib(
           ^^^^^^^^^^^^^^^^^^
  File "D:\04_GitRepos\TEASER\teaser\project.py", line 1095, in export_aixlib
    model_report.calc_report_data(prj=self, path=report_path)
  File "D:\04_GitRepos\TEASER\teaser\data\output\reports\model_report.py", line 220, in calc_report_data
    window_area_north=prj_data_flat["Window Area_0.0"],
                      ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
KeyError: 'Window Area_0.0'`

File: teaser/data/output/reports/model_report.py (hunk 1 above)
Comment (DaJansenGit) [reply teaser/data/output/reports/model_report.py:225]: roofs are on #todo, this is also noted in the code.

**Review by MartinRaetz — CHANGES_REQUESTED** [review]
Upon using other building types the following error occurs and suggests that non existent walls/windows need to be handled:
This test used example 6 to be exported.

`Traceback (most recent call last):
  File "D:\04_GitRepos\TEASER\teaser\examples\e2_export_aixlib_models.py", line 81, in <module>
    example_export_aixlib()
  File "D:\04_GitRepos\TEASER\teaser\examples\e2_export_aixlib_models.py", line 71, in example_export_aixlib
    path = prj.export_aixlib(
           ^^^^^^^^^^^^^^^^^^
  File "D:\04_GitRepos\TEASER\teaser\project.py", line 1095, in export_aixlib
    model_report.calc_report_data(prj=self, path=report_path)
  File "D:\04_GitRepos\TEASER\teaser\data\output\reports\model_report.py", line 220, in calc_report_data
    window_area_north=prj_data_flat["Window Area_0.0"],
                      ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
KeyError: 'Window Area_0.0'`


When handling this error, I suggest to think about the following possibilities as well:

1. Model does not have outer walls to all directions (or windows)
2. Model has no outerroof, no roof, no groundfloor or no floor.
3. One "wall" is entirely windows
4. Roof tilt (is this currently displayed?)
5. Wall direction is 34° (something else than each 90°)

I do think it is important that the code will run through in any case, though, I do not think it is important that every possibility is displayable in the 3D Plot. Depending on the effort feel free to decide which configurations shall be supported and which will just result in a blank or ?errorenous? 3D plot.
Possibly also add the information, that every building is displayed as square, other shapes will be transformed or dismissed.

**Comment by DaJansenGit** [comment]
> Upon using other building types the following error occurs and suggests that non existent walls/windows need to be handled: This test used example 6 to be exported.
> 
> `Traceback (most recent call last): File "D:\04_GitRepos\TEASER\teaser\examples\e2_export_aixlib_models.py", line 81, in <module> example_export_aixlib() File "D:\04_GitRepos\TEASER\teaser\examples\e2_export_aixlib_models.py", line 71, in example_export_aixlib path = prj.export_aixlib( ^^^^^^^^^^^^^^^^^^ File "D:\04_GitRepos\TEASER\teaser\project.py", line 1095, in export_aixlib model_report.calc_report_data(prj=self, path=report_path) File "D:\04_GitRepos\TEASER\teaser\data\output\reports\model_report.py", line 220, in calc_report_data window_area_north=prj_data_flat["Window Area_0.0"], ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^ KeyError: 'Window Area_0.0'`
> 
> When handling this error, I suggest to think about the following possibilities as well:
> 
> 1. Model does not have outer walls to all directions (or windows)
> 2. Model has no outerroof, no roof, no groundfloor or no floor.
> 3. One "wall" is entirely windows
> 4. Roof tilt (is this currently displayed?)
> 5. Wall direction is 34° (something else than each 90°)
> 
> I do think it is important that the code will run through in any case, though, I do not think it is important that every possibility is displayable in the 3D Plot. Depending on the effort feel free to decide which configurations shall be supported and which will just result in a blank or ?errorenous? 3D plot. Possibly also add the information, that every building is displayed as square, other shapes will be transformed or dismissed.

Thanks for your feedback. I put some changes to this with b65c15d231f811ddb56d4645d2cced56b122bb8d and solves some of the problems. But their are still some limitations which are described in the docstrings:
- no roof support yet (angle is just a dummy)
- only support of buildings with 4 orientations (but it can be any orienation). But the building will not be rotated yet. Meaning that the smallest orientation value  (0-359°) will be taken as north, the next as east and so on.

As I can't put much more time in this right now I will leave it at this for now.

**Review by MartinRaetz — DISMISSED** [review]
Thank you for that nice feature!

**Review by MartinRaetz — APPROVED** [review]

**Review by MartinRaetz — APPROVED** [review]
Even better

**Pull request merged**